### DISCUSSION TITLE
Rendering images and drawing annotations

### DISCUSSION THREAD

**Discussion by mdylan2** [observablehq/plot#2223] [Q&A] [answered]
Is there a way to plot images, and make annotations on top of them (eg: drawing bounding boxes)?

Imshow example: https://www.geeksforgeeks.org/matplotlib-pyplot-imshow-in-python/

**Comment by Fil** [comment, marked as answer]
Yes! Combine a [raster](https://observablehq.com/plot/marks/raster) mark and a [rect](https://observablehq.com/plot/marks/rect) mark.

**Reply by mdylan2** [reply]
How would this work for RGB images? Is it one raster per channel? 

Never mind, figured it out, I think you can just multiply the length of the data in the grid that you put in

**Comment by mdylan2** [comment]
This makes sense! What about free-form annotation? 

I'm working with images of bacterial cells, which are oddly shaped - how would you handle event driven free-form annotation?

**Reply by Fil** [reply]
To follow the pointer and clicks, you can use https://observablehq.com/@fil/pointer-events-svg

However you'll need a system to record the shapes. You could save them to localStorage for example, or allow the user to download them as a json file.

You can then display the shapes with Plot.geo
